crates/tabby/src/services/code.rs

```diff
@@ -151,27 +152,40 @@ fn closest_match<'a>(
     search_term: &'a str,
     search_input: impl IntoIterator<Item = &'a RepositoryConfig>,
 ) -> Option<String> {
-    let search_term = RepositoryConfig::canonicalize_url(search_term);
-
-    let mut nucleo = nucleo::Matcher::new(nucleo::Config::DEFAULT.match_paths());
+    let git_search = GitUrl::parse(search_term).ok()?;
     search_input
         .into_iter()
-        .filter_map(|entry| {
-            let url = entry.canonical_git_url();
-            Some((
-                url.clone(),
-                // Matching using the input URL as the haystack instead of the needle yielded better scoring
-                // Example:
-                // haystack = "https://github.com/boxbeam/untwine" needle = "https://abc@github.com/boxbeam/untwine.git" => No match
-                // haystack = "https://abc@github.com/boxbeam/untwine.git" needle = "https://github.com/boxbeam/untwine" => Match, score 842
-                nucleo.fuzzy_match(
-                    Utf32String::from(&*search_term).slice(..),
-                    Utf32String::from(&*url).slice(..),
-                )?,
-            ))
+        .filter_map(|elem| {
+            let Ok(git_url) = GitUrl::parse(&elem.git_url) else {
```

Comment:
what if url doesn't come with owner? can you add a test case for it? e.g `http://abc.com/def`